src/blog/2023/09/making-sense-of-your-industrial-data.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Making sense of your industrial data
+subtitle: A look at legacy industrial protocol data and making sense of it in an IIoT world.
+description: A look at legacy industrial protocol data and making sense of it in an IIoT world.
+date: 2023-09-08
+authors: ["stephen-mclaughlin"]
+image: "/blog/2023/09/images/industrial-legacy-data-blog-image.png"
+tags:
+    - posts
+    - node-red
+    - community
+---
+
+In today's rapidly evolving industrial landscape, the ability to collect, analyse, and act upon data is more critical than ever. The Industrial Internet of Things (IIoT) has ushered in a new era of connectivity and automation, but what about the wealth of data locked away in legacy industrial protocols like ModBus? How can we unlock and make sense of this valuable information in the IIoT era? That's where tools like Node-RED and node-red-contrib-buffer-parser come into play.
+
+<!--more-->
+
+### Legacy Industrial Data: ModBus
+
+ModBus is one of the most widely used industrial protocols. Originally developed in the late 1970s, it has been a popular choice for industrial communication ever since. However, its data format can be challenging to work with in the context of modern IIoT applications. ModBus typically represents data in 16-bit unsigned registers, making it necessary to convert this data into more usable formats like Signed integer, Float, Signed and Unsigned Long, String, or even individual bits.
+
+### A short primer on data types
+
+Before we dive into how to make sense of ModBus data, let's take a quick look at some of the common data types we have to deal with.
+
+#### 16-bit unsigned
+
+16-bit unsigned data is an integer that can only be positive. It can represent values from 0 to 65535. For example, the number 12345 is represented as `0x3039` in hexadecimal or `0011000000111001` in binary.
+
+#### 16-bit signed
+
+16-bit signed data is an integer that can be positive or negative. It can represent values from -32768 to 32767. For example, the number -12345 is represented as `0xCFC7` in hexadecimal or `1100111111000111` in binary.
```

Comment:
So now we need a bit to denote positive or negative, and so the number of values we can represent needs to be stored in 15 bits. So the maximum value is 2^15 -1

Reply:
As previous comment, I would much prefer to ignore scientific notation and gently introduce the reader to everyday constants. If you dont mind, I would prefer to leave as is.